src/query/sql/src/executor/format.rs

```diff
@@ -152,7 +171,14 @@ fn filter_to_format_tree(plan: &Filter, metadata: &MetadataRef) -> Result<Format
         children.extend(items);
     }
 
-    children.push(to_format_tree(&plan.input, metadata)?);
+    if let Some(prof_span) = prof_span_set.lock().unwrap().get(&plan.plan_id) {
+        let process_time = prof_span.process_time / 1000 / 1000; // milliseconds
+        children.push(FormatTreeNode::new(format!(
+            "total process time: {process_time}ms"
```

Comment:
Can we add `total process time` to `TableScan` as well?
This explains analyze the cost of 16.39sec, most cost is TableScan:
```
mysql> explain analyze SELECT * FROM hits WHERE URL LIKE '%google%' ORDER BY EventTime LIMIT 10;
+----------------------------------------------------------------------------------------------------------------------------------------+
| explain                                                                                                                                |
+----------------------------------------------------------------------------------------------------------------------------------------+
| Limit                                                                                                                                  |
| ├── limit: 10                                                                                                                          |
| ├── offset: 0                                                                                                                          |
| ├── estimated rows: 10.00                                                                                                              |
| ├── total process time: 0ms                                                                                                            |
| └── Sort                                                                                                                               |
|     ├── sort keys: [eventtime ASC NULLS LAST]                                                                                          |
|     ├── estimated rows: 33332499.00                                                                                                    |
|     ├── total process time: 157ms                                                                                                      |
|     └── Filter                                                                                                                         |
|         ├── filters: [like(hits.url (#13), "%google%")]                                                                                |
|         ├── estimated rows: 33332499.00                                                                                                |
|         ├── total process time: 4658ms                                                                                                 |
|         └── TableScan                                                                                                                  |
|             ├── table: default.hits.hits                                                                                               |
|             ├── read rows: 99997497                                                                                                    |
|             ├── read bytes: 15458917407                                                                                                |
|             ├── partitions total: 751                                                                                                  |
|             ├── partitions scanned: 751                                                                                                |
|             ├── pruning stats: [segments: <range pruning: 32 to 32>, blocks: <range pruning: 751 to 751, bloom pruning: 0 to 0>]       |
|             ├── push downs: [filters: [like(hits.url (#13), "%google%")], limit: NONE]                                                 |
|             └── estimated rows: 99997497.00                                                                                            |
+----------------------------------------------------------------------------------------------------------------------------------------+
22 rows in set (16.39 sec)
Read 99997497 rows, 75.78 GiB in 16.231 sec., 6.16 million rows/sec., 4.67 GiB/sec.
```

Reply:
We can do that, but that will make a lot of modifications to the `Table::read_data`. 

This PR is kind of an experimental implementation, which won't involve the full features of query profiling.

By the way, I think we should measure the cost of the table scans with some I/O metrics instead of `process time` since they are always asynchronous.

Reply:
For I/O metrics, we can add more to https://github.com/datafuselabs/databend/blob/80b5d2d403e9aa31d72ed5b281121ce105bec58d/src/query/catalog/src/plan/pruning_statistics.rs#L16-L28